python/whylogs/core/view/segmented_dataset_profile_view.py

```diff
@@ -0,0 +1,184 @@
+                f.write(chunk_msg.SerializeToString())
+
+            total_len = f.tell()
+            f.flush()
+
+            # calculate segment tags based on columnar segments
+            segment_message_tags, segment_tags, segment_metadata = _generate_segment_tags_metadata(
+                self.segment, self.partition
+            )
+
+            if not segment_tags:
+                raise NotImplementedError(
+                    f"Serialization of segments requires segment tags but none calculated for partition: {self.partition} and segments: {self.segment}"
+                )
+
+            segment_message = SegmentMessage()
+            segment_message.tags.extend(segment_tags)
+            segments_message_field = [segment_message]
+
+            properties = DatasetProperties(
+                dataset_timestamp=to_utc_milliseconds(self.profile_view._dataset_timestamp),
+                creation_timestamp=to_utc_milliseconds(self.profile_view._creation_timestamp),
+                metadata=segment_metadata,
+                tags=segment_message_tags,
+            )
+
+            logger.warn(f"constructed DatasetProperties for segmented profile file: {properties}")
+
+            dataset_header = DatasetProfileHeader(
+                column_offsets=column_chunk_offsets,
+                properties=properties,
+                length=total_len,
+                indexed_metric_paths=metric_index_to_name,
+            )
+
+            # TODO: multi segment file format requires multiple offset calculations.
+            segment_offsets: Dict[int, int] = {
+                n: 0 for n in range(len(segments_message_field))
+            }  # update this after writing
+
+            # single file segments.
+            dataset_segment_header = DatasetSegmentHeader(has_segments=True, offsets=segment_offsets)
+
+            # TODO: calculate other segments offsets when we support multiple segments per file
+            write_delimited_protobuf(f, dataset_segment_header)
+            first_segment_offset = f.tell() - total_len
+            dataset_segment_header.offsets[0] = first_segment_offset
+
+            # only single segment files at first.
+            dataset_segment_header.segments.extend(segments_message_field)
+
+            with open(path, "w+b") as out_f:
+                out_f.write(WHYLOGS_MAGIC_HEADER_BYTES)
+                write_delimited_protobuf(out_f, dataset_segment_header)
+                logger.debug(
+                    f"Writing segmented profile file: whylogs file and segment headers wrote {out_f.tell()} bytes"
+                )
+                write_delimited_protobuf(out_f, dataset_header)
+                logger.debug(
+                    f"Writing segmented profile file: and with dataset header wrote a total of {out_f.tell()} bytes before writing the chunks."
+                )
+
+                f.seek(0)
+                while f.tell() < total_len:
+                    buffer = f.read(1024)
```

Comment:
This is true for here and DatasetProfileView, let's make this not a magic number for easier change.

Reply:
Agreed, might defer to keep the scope limited to not fixing DatasetProfileView in this PR?

Reply:
Created a constant for this file, but leaving DatasetProfileView out of this change for now.